openmc/arithmetic.py

```diff
@@ -0,0 +1,842 @@
+                  '"{0}" is not one of the bins'.format(filter_bin)
+            raise ValueError(msg)
+        else:
+            return self.bins.index(filter_bin)
+
+    def get_pandas_dataframe(self, data_size, stride, summary=None, **kwargs):
+        """Builds a Pandas DataFrame for the AggregateFilter's bins.
+
+        This method constructs a Pandas DataFrame object for the AggregateFilter
+        with columns annotated by filter bin information. This is a helper
+        method for the Tally.get_pandas_dataframe(...) method.
+
+        Parameters
+        ----------
+        data_size : int
+            The total number of bins in the tally corresponding to this filter
+        stride : int
+            Stride in memory for the filter
+        summary : None or Summary
+            An optional Summary object to be used to construct columns for
+            distribcell tally filters (default is None). NOTE: This parameter
+            is not used by the AggregateFilter and simply mirrors the method
+            signature for the CrossFilter.
+
+        Returns
+        -------
+        pandas.DataFrame
+            A Pandas DataFrame with columns of strings that characterize the
+            aggregatefilter's bins. Each entry in the DataFrame will include
+            one or more aggregation operations used to construct the
+            aggregatefilter's bins. The number of rows in the DataFrame is the
+            same as the total number of bins in the corresponding tally, with
+            the filter bins appropriately tiled to map to the corresponding
+            tally bins.
+
+        See also
+        --------
+        Tally.get_pandas_dataframe(), Filter.get_pandas_dataframe(),
+        CrossFilter.get_pandas_dataframe()
+
+        """
+        # Create NumPy array of the bin tuples for repeating / tiling
+        filter_bins = np.empty(self.num_bins, dtype=tuple)
+        for i, bin in enumerate(self.bins):
+            filter_bins[i] = bin
+
+        # Repeat and tile bins as needed for DataFrame
+        filter_bins = np.repeat(filter_bins, stride)
+        tile_factor = data_size / len(filter_bins)
+        filter_bins = np.tile(filter_bins, tile_factor)
+
+        # Create DataFrame with aggregated bins
+        df = pd.DataFrame({self.type: filter_bins})
+        return df
+
+    def can_merge(self, other):
+        """Determine if AggregateFilter can be merged with another.
+
+        Parameters
+        ----------
+        other : AggregateFilter
+            Filter to compare with
+
+        Returns
+        -------
+        bool
+            Whether the filter can be merged
+
+        """
+
+        if not isinstance(other, AggregateFilter):
+            return False
+
+        # Filters must be of the same type
+        elif self.type != other.type:
+            return False
+
+        # None of the bins in this filter should match in the other filter
+        for bin in self.bins:
```

Comment:
A tiny bit wordier, but also cleaner, I might opt to use an "any" here
```
return any(bin in other.bins for bin in self.bins)
```
technically this cuts 5 lines down to 1